datafusion/expr/src/logical_plan/invariants.rs

```diff
@@ -281,13 +383,30 @@ fn check_mixed_out_refer_in_window(window: &Window) -> Result<()> {
     }
 }
 
+fn collect_subquery_cols(
```

Comment:
this seems like a copy of the code in https://github.com/apache/datafusion/blob/344f0897d105cdd588d7a7ee8cef0646f844937a/datafusion/optimizer/src/utils.rs#L90-L89

It would be nice if it were shared -- perhaps it could be moved in https://github.com/apache/datafusion/blob/2ac8af894f9aefb3215b74044b7f177b42f7e65f/datafusion/expr/src/utils.rs#L18-L17 to be shared